spring-xd-shell/src/test/java/org/springframework/xd/shell/command/FileSourceAndFileSinkTests.java

```diff
@@ -113,4 +120,38 @@ private void testFileSinkSource(String sinkSuffix, String sinkParam) throws Exce
 			out.delete();
 		}
 	}
+
+	private void testFileSinkSource(String nameExpressionParam, String dirPrefix, String dirSuffix) throws Exception {
+		// Are we on *nix at least?
+		if (!new File("/tmp/").exists()) {
+			return;
+		}
+
+		// Both use stream name
+		String streamName = generateStreamName();
+
+		File inDir = new File(DEFAULT_IN, streamName);
+		inDir.mkdirs();
+		File in = new File(inDir, "one.txt");
+		File outDir = new File("/tmp/" + dirPrefix + "hello" + dirSuffix);
+		File out = new File(outDir, "hello");
+
+		try {
+			FileCopyUtils.copy("hello", new FileWriter(in));
+			String directoryExpression = " --directoryExpression='''/tmp/'' + ''" + dirPrefix + "'' + payload.trim() + ''"
+					+ dirSuffix + "'''";
+			stream().create(streamName, "in: file --outputType=text/plain | out: file " + nameExpressionParam +
+					directoryExpression);
+			Thread.sleep(1000);
+			String actual = FileCopyUtils.copyToString(new FileReader(out));
+			assertEquals("hello", actual.trim());
+			assertEquals(out.exists(), is(true));
```

Comment:
@garyrussel : yes it should be assertThat ... and that's what I'm used to use ... my mistake ! But what bothers me the most : why I did not get exception when I ran the gradlew dist ? I missed something ...